Tests/Common/Securities/CashAccountMarginModelTests.cs

```diff
@@ -0,0 +1,326 @@
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.BTCUSD, Resolution.Minute, tz, tz, true, false, false),
+                SymbolProperties.GetDefault(CashBook.AccountCurrency));
+
+            _ethusd = new Crypto(
+                SecurityExchangeHours.AlwaysOpen(tz),
+                _portfolio.CashBook[CashBook.AccountCurrency],
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.ETHUSD, Resolution.Minute, tz, tz, true, false, false),
+                SymbolProperties.GetDefault(CashBook.AccountCurrency));
+
+            _btceur = new Crypto(
+                SecurityExchangeHours.AlwaysOpen(tz),
+                _portfolio.CashBook["EUR"],
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.BTCEUR, Resolution.Minute, tz, tz, true, false, false),
+                SymbolProperties.GetDefault("EUR"));
+
+            _ethbtc = new Crypto(
+                SecurityExchangeHours.AlwaysOpen(tz),
+                _portfolio.CashBook["BTC"],
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.ETHBTC, Resolution.Minute, tz, tz, true, false, false),
+                SymbolProperties.GetDefault("BTC"));
+
+            _marginModel = new CashAccountMarginModel(_portfolio.CashBook);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            _transactionHandler.Exit();
+        }
+
+        [Test]
+        public void InitializesCorrectly()
+        {
+            Assert.AreEqual(1m, _marginModel.GetLeverage(_spy));
+            Assert.AreEqual(1m, _marginModel.GetInitialMarginRequirement(_spy));
+            Assert.AreEqual(0m, _marginModel.GetMaintenanceMargin(_spy));
+            Assert.AreEqual(0m, _marginModel.GetMaintenanceMarginRequirement(_spy));
+        }
+
+        [Test]
+        public void SetLeverageDoesNotUpdateLeverage()
+        {
+            // leverage cannot be set, will always be 1x
+            _marginModel.SetLeverage(_spy, 50m);
+            Assert.AreEqual(1m, _marginModel.GetLeverage(_spy));
+        }
+
+        [Test]
+        public void EquityBuyOrderRequiresFullMargin()
+        {
+            _spy.SetMarketPrice(new Tick { Value = 250m });
+
+            var order = new MarketOrder(_spy.Symbol, 2, DateTime.UtcNow);
+            var initialMargin = _marginModel.GetInitialMarginRequiredForOrder(_spy, order);
+
+            Assert.AreEqual(501m, initialMargin);
+        }
+
+        [Test]
+        public void EquitySellOrderRequiresHolding()
+        {
+            _spy = _algorithm.AddEquity("SPY");
+            _portfolio[_spy.Symbol].SetHoldings(150, 2);
+            _spy.SetMarketPrice(new Tick { Value = 250m });
+            _algorithm.SetFinishedWarmingUp();
+
+            var order = new MarketOrder(_spy.Symbol, -2, DateTime.UtcNow);
+            var initialMargin = _marginModel.GetInitialMarginRequiredForOrder(_spy, order);
+
+            Assert.AreEqual(-499m, initialMargin);
```

Comment:
I'm thinking that in the sell case we should be tracking units of stock (or currency/whatever). The model is really saying you can sell 2 units, and that's true even if the price moves quickly.

Same here for comment, how did you get -499?